src/main/extension-filesystem.ts

```diff
@@ -0,0 +1,58 @@
+import { randomBytes } from "crypto"
+import { SHA256 } from "crypto-js"
+import { app } from "electron"
+import { ensureDir, mkdir, pathExists } from "fs-extra"
+import { action, observable, toJS } from "mobx"
+import path from "path"
+import { BaseStore } from "../common/base-store"
+
+interface FSProvisionModel {
+  extensions: Record<string, string>; // extension names to paths
+}
+
+export class FilesystemProvisionerStore extends BaseStore<FSProvisionModel> {
+  @observable registeredExtensions = new Map<string, string>()
+
+  private constructor() {
+    super({
+      configName: "lens-filesystem-provisioner-store",
+      accessPropertiesByDotNotation: false, // To make dots safe in cluster context names
+    });
+  }
+
+  /**
+   * This function retrieves the saved path to the folder which the extension
+   * can saves files to. If the folder is not present then it is created.
+   * @param extensionName the name of the extension requesting the path
+   * @returns path to the folder that the extension can safely write files to.
+   */
+  async requestDirectory(extensionName: string): Promise<string> {
+    if (!this.registeredExtensions.has(extensionName)) {
+      const salt = randomBytes(32).toString("hex")
+      const hashedName = SHA256(`${extensionName}/${salt}`).toString()
```

Comment:
Looks overcomplicated. Why all this hashes needed? For security?

Reply:
I mentioned this in my original post. Maybe I should copy some of that here.... Namely, I want to force extension writers to use the methods provided in this PR and I don't want the name of the folder to be per-determinable. That way it is harder (but not impossible since we are doing no actual sand-boxing) for a malicious extension to find the folder of a specific other extension.

Reply:
Okay, sounds reasonable.